beacon-chain/blockchain/kzg/kzg.go

```diff
@@ -108,39 +109,48 @@ func VerifyCellKZGProofBatch(commitmentsBytes []Bytes48, cellIndices []uint64, c
 	return ckzg4844.VerifyCellKZGProofBatch(commitmentsBytes, cellIndices, ckzgCells, proofsBytes)
 }
 
-// RecoverCellsAndKZGProofs recovers the complete cells and KZG proofs from a given set of cell indices and partial cells.
+// RecoverCells recovers the complete cells from a given set of cell indices and partial cells.
 // Note: `len(cellIndices)` must be equal to `len(partialCells)` and `cellIndices` must be sorted in ascending order.
-func RecoverCellsAndKZGProofs(cellIndices []uint64, partialCells []Cell) (CellsAndProofs, error) {
+func RecoverCells(cellIndices []uint64, partialCells []Cell) ([]Cell, error) {
 	// Convert `Cell` type to `ckzg4844.Cell`
 	ckzgPartialCells := make([]ckzg4844.Cell, len(partialCells))
 	for i := range partialCells {
 		ckzgPartialCells[i] = ckzg4844.Cell(partialCells[i])
 	}
 
-	ckzgCells, ckzgProofs, err := ckzg4844.RecoverCellsAndKZGProofs(cellIndices, ckzgPartialCells)
+	ckzgCells, err := ckzg4844.RecoverCells(cellIndices, ckzgPartialCells)
 	if err != nil {
-		return CellsAndProofs{}, errors.Wrap(err, "recover cells and KZG proofs")
+		return nil, errors.Wrap(err, "recover cells")
+	}
+
+	cells := make([]Cell, len(ckzgCells))
+	for i := range ckzgCells {
+		cells[i] = Cell(ckzgCells[i])
 	}
 
-	return makeCellsAndProofs(ckzgCells[:], ckzgProofs[:])
+	return cells, nil
 }
 
-// makeCellsAndProofs converts cells/proofs to the CellsAndProofs type defined in this package.
-func makeCellsAndProofs(ckzgCells []ckzg4844.Cell, ckzgProofs []ckzg4844.KZGProof) (CellsAndProofs, error) {
-	if len(ckzgCells) != len(ckzgProofs) {
-		return CellsAndProofs{}, errors.New("different number of cells/proofs")
+// RecoverCellsAndKZGProofs recovers the complete cells and KZG proofs from a given set of cell indices and partial cells.
+// Note: `len(cellIndices)` must be equal to `len(partialCells)` and `cellIndices` must be sorted in ascending order.
+func RecoverCellsAndKZGProofs(cellIndices []uint64, partialCells []Cell) ([]Cell, []Proof, error) {
+	// Convert `Cell` type to `ckzg4844.Cell`
+	ckzgPartialCells := make([]ckzg4844.Cell, len(partialCells))
+	for i := range partialCells {
+		ckzgPartialCells[i] = ckzg4844.Cell(partialCells[i])
 	}
 
-	cells := make([]Cell, 0, len(ckzgCells))
-	proofs := make([]Proof, 0, len(ckzgProofs))
+	ckzgCells, ckzgProofs, err := ckzg4844.RecoverCellsAndKZGProofs(cellIndices, ckzgPartialCells)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "recover cells and KZG proofs")
+	}
 
+	cells := make([]Cell, len(ckzgCells))
+	proofs := make([]Proof, len(ckzgProofs))
 	for i := range ckzgCells {
-		cells = append(cells, Cell(ckzgCells[i]))
-		proofs = append(proofs, Proof(ckzgProofs[i]))
+		cells[i] = Cell(ckzgCells[i])
+		proofs[i] = Proof(ckzgProofs[i][:])
```

Comment:
Will panic if `len(ckzgCells) > len(ckzgProofs)`.
I know there is a warning about this in the godoc. But IMO, we should:
- either modify the godoc specifically indicating that the function will panic if array do not have the same size. (Because the function returns an error, the user may think an error will be returned in such a case, not a panic), or
- re-introduce the very cheap `len` check. 

I prefer the second solution.